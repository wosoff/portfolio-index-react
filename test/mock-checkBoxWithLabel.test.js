import React from 'react'
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from '@testing-library/react'

import MockCheckboxWithLabel from './mock/mock-checkboxWithLabel'

describe('mock_checkboxWithLabe', () => {
  afterEach(cleanup)

  test.skip('test with react-test-renderer', () => {
  // const Component = renderer.create(<CheckboxWithLabel labelOn="On" labelOff="Off" />)

  // let tree = Component.toJSON();

  // tree.props.onMouseEnter();
  // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
  })


  test('changes the text after click', () => {
    const {queryByLabelText, getByLabelText, container} = render(
      <MockCheckboxWithLabel labelOn="On" labelOff="Off" />,
    );

    expect(container).toMatchSnapshot()
    
    expect(queryByLabelText(/off/i)).toBeTruthy();

    fireEvent.click(getByLabelText(/off/i));

    expect(queryByLabelText(/on/i)).toBeTruthy();
  })
})
