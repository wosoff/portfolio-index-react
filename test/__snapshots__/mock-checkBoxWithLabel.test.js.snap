// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock_checkboxWithLabe changes the text after click 1`] = `
<div>
  <label>
    <input
      type="checkbox"
    />
    Off
  </label>
</div>
`;
